x/wasm/keeper/gas_register.go

```diff
@@ -0,0 +1,128 @@
+package keeper
+
+import (
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	// DefaultGasMultiplier is how many cosmwasm gas points = 1 sdk gas point
+	// SDK reference costs can be found here: https://github.com/cosmos/cosmos-sdk/blob/02c6c9fafd58da88550ab4d7d494724a477c8a68/store/types/gas.go#L153-L164
+	// A write at ~3000 gas and ~200us = 10 gas per us (microsecond) cpu/io
+	// Rough timing have 88k gas at 90us, which is equal to 1k sdk gas... (one read)
+	//
+	// Please note that all gas prices returned to the wasmer engine should have this multiplied
+	DefaultGasMultiplier uint64 = 100
+	// DefaultInstanceCost is how much SDK gas we charge each time we load a WASM instance.
+	// Creating a new instance is costly, and this helps put a recursion limit to contracts calling contracts.
+	DefaultInstanceCost uint64 = 40_000
+	// DefaultCompileCost is how much SDK gas we charge *per byte* for compiling WASM code.
+	DefaultCompileCost uint64 = 2
+	// DefaultEventAttributeDataCost is how much SDK gas we charge *per byte* for attribute data in events.
+	// This is len(key) + len(value)
+	DefaultEventAttributeDataCost uint64 = 1
+	// DefaultPerAttributeCost is how much SDK gas we charge per attribute count.
+	DefaultPerAttributeCost uint64 = 10
+	// DefaultEventAttributeDataFreeTier number of bytes of attribute data we do not charge.
+	DefaultEventAttributeDataFreeTier = 100
+)
+
+type GasRegister struct {
+	instanceCost  sdk.Gas
+	compileCost   sdk.Gas
+	gasMultiplier sdk.Gas
+
+	eventPerAttributeCost      sdk.Gas
+	eventAttributeDataCost     sdk.Gas
+	eventAttributeDataFreeTier int
+}
+
+func DefaultGasRegister() GasRegister {
+	return GasRegister{
+		instanceCost:               DefaultInstanceCost,
+		compileCost:                DefaultCompileCost,
+		gasMultiplier:              DefaultGasMultiplier,
+		eventPerAttributeCost:      DefaultPerAttributeCost,
+		eventAttributeDataCost:     DefaultEventAttributeDataCost,
+		eventAttributeDataFreeTier: DefaultEventAttributeDataFreeTier,
+	}
+}
+func NewGasRegister(
```

Comment:
This can get quite confusing to construct with 5 items of the same type. Easy to mix up order instance and compile cost for example.

In such cases, I find making the members public and just constructing them like:

```golang
gas := GasRegister {
  InstanceCost: instance,
  CompileCost: compile,
  // ...
}
```

is more robust and more readable.

Reply:
Good point. I was a bit afraid of updates from outside the package but I will make a config object with public attributes so that we can have both.